WcaOnRails/app/views/competitions/my_competitions.html.erb

```diff
@@ -8,29 +8,97 @@
+            <% message = "This competition is confirmed but not visible." %>
+          <% end %>
+        <% else %>
+          <% if competition.showAtAll %>
+            <% message = "This competition is not confirmed but is visible." %>
+          <% else %>
+            <% message = "This competition is not confirmed and not visible." %>
+          <% end %>
         <% end %>
-      <% end %>
-      <% table.column do |competition| %>
-        <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
+        <tr class="<%= competition.isConfirmed ? "confirmed" : "not-confirmed" %> <%= competition.showAtAll ? "visible" : "not-visible" %>" data-toggle="tooltip" data-placement="top" data-container="body" title="<%= message %>">
+          <td><%= link_to competition.name, competition_path(competition) %></td>
+          <td><%= competition.cityName %>, <%= competition.country_name %></td>
+          <td><%= competition.start_date %></td>
+          <td>
+          <% if current_user.can_manage_competition?(competition) %>
+            <%= link_to "Edit", edit_competition_path(competition) %>
+          <% end %>
+          </td>
+          <td>
+            <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
           <%= link_to "Registrations", competition_edit_registrations_path(competition) %>
         <% end %>
+          </td>
+        </tr>
       <% end %>
-    <% end %>
-  <% end %>
+    </table>
 
-  <% if current_user.wca_id %>
-    <%= link_to "My profile", "/results/p.php?i=" + current_user.wca_id %>
+    <%= button_tag("Show past competitions", id: "show-hidden", class: "btn btn-primary") %>
   <% end %>
 
+  <div id="past-competitions">
+    <h2>Past competitions</h2>
+    <table class="table competitions-table">
+      <tr>
+        <th>Name</th>
+        <th>Location</th>
+        <th>Start date</th>
+        <th></th>
+        <th></th>
+      </tr>
+      <% @past_competitions.each do |competition| %>
+        <tr>
+          <td><%= link_to competition.name, competition_path(competition) %></td>
+          <td><%= competition.cityName %>, <%= competition.country_name %></td>
+          <td><%= competition.start_date %></td>
+          <td>
+          <% if current_user.can_manage_competition?(competition) %>
+            <%= link_to "Edit", edit_competition_path(competition) %>
+          <% end %>
+          </td>
+          <td>
+            <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
+          <%= link_to "Registrations", competition_edit_registrations_path(competition) %>
+        <% end %>
+          </td>
+        </tr>
+      <% end %>
+    </table>
+  </div>
+
+  <p>
+    <% if current_user.wca_id %>
+      <%= link_to "My profile", "/results/p.php?i=" + current_user.wca_id %>
+    <% end %>
+  </p>
+
 </div>
+
+<script>
+  (function() {
+    $("#past-competitions").hide();
```

Comment:
Can we get rid of this and use bootstrap's collapse instead? See http://getbootstrap.com/javascript/#collapse